packages/components/src/components/modal/modal.tsx

```diff
@@ -174,6 +176,10 @@ export class Modal {
     this.hasHeader = hasHeading(this.host, this.heading) || hasNamedSlot(this.host, 'header');
     this.hasFooter = hasNamedSlot(this.host, 'footer');
 
+    if (this.hasHeader) {
```

Comment:
Same as `flyout`. Instead of `ariaLabelText` directly use function which returns ariaLabelText